src/aosm/azext_aosm/configuration_models/onboarding_sns_input_config.py

```diff
@@ -19,21 +19,19 @@ class SiteNetworkServicePropertiesConfig:
             "comment": "Azure location to use when creating resources e.g uksouth"
         },
     )
-    operator_resource_group: str = field(
+    operator_resource_group_name: str = field(
         default="",
         metadata={"comment": "The resource group that the operator resources will be hosted in."},
     )
     site_name: str = field(
         default="",
         metadata={"comment": "Name of the site."},
     )
-    nsd_reference: "NSDVReferenceConfig" = (
+    nsd_reference: NSDVReferenceConfig = (
```

Comment:
If you delete the nsd_reference completely and run this, the error is "Publisher Name must be set". I think it should instead be "You must include at least one nsd_reference". I'm not sure if the rest of our code behaves like this, so maybe we can raise a more general story to fix the validation for objects in the config if not

Reply:
others also behave this way, will take up fixing this next